server/xpub-controller/helpers/files.js

```diff
@@ -0,0 +1,161 @@
+const logger = require('@pubsweet/logger')
+const FileModel = require('@elifesciences/xpub-model').File
+
+class FilesHelper {
+  constructor(config, scienceBeamApi) {
+    this.config = config
+    this.scienceBeamApi = scienceBeamApi
+  }
+
+  validateFileSize(fileSize) {
+    if (fileSize > this.config.get('fileUpload.maxSizeMB') * 1e6) {
+      throw new Error(
+        `File size shouldn't exceed ${this.config.get(
+          'fileUpload.maxSizeMB',
+        )}MB`,
+      )
+    }
+  }
+
+  static async generateFileEntity(file, manuscriptId) {
+    const { stream, filename, mimetype: mimeType } = await file
+
+    const fileEntity = await new FileModel({
+      manuscriptId,
+      url: `manuscripts/${manuscriptId}`,
+      filename,
+      type: 'MANUSCRIPT_SOURCE_PENDING',
+      mimeType,
+    }).save()
+    return {
+      stream,
+      fileEntity,
+    }
+  }
+
+  static publishPredictedProgress(
```

Comment:
Is it worth renaming this to `publishPredictedProgressThunk`?

We're expecting to delay the execution of this function, so it may be worth making that clearer.

And if we are using thunks, the validation usually happens within the function, and the wrapping function is left very thin. 

You can read more about thunks [here](https://en.wikipedia.org/wiki/Thunk)

Reply:
Actually, I don't have enough experience with thunks to recommend them here. I'd say avoid them unless someone else wants to give their input on the topic.